<div style="text-align: center; padding: 35px; background-color: #f5f7fa; border-radius: 16px; box-shadow: 0px 6px 14px rgba(0,0,0,0.1); margin-bottom: 30px;">

  <h1 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:40px; margin-bottom:10px;">
    Functional Question 1: NR1 Loop Phosphosite Analysis
  </h1>

  <h3 style="color:#7f8c8d; font-family:'Georgia',serif; font-weight:normal; font-size:22px; margin-top:0;">
    Phosphosite Analysis of NR1
  </h3>

  <p style="color:#34495e; font-family:'Arial',sans-serif; font-size:16px; margin-top:20px;">
    Author: <b>Vallari Eastman</b> <br>
    Date: <b>Dec 2025</b>
  </p>
</div>


<div style="background-color:#f8f9fb; padding:20px; border-radius:10px;">
    <div style="background-color:#f8f9fb; padding:20px; border-radius:10px;">
    <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    Formal Functional Question 1:
  </h2>
<p style="color: #34495e; font-family: 'Arial', sans-serif; font-size: 15px; margin-top: 18px;">
        Do predicted phosphosites co-localize with flexible loops that regulate epitope masking?
 </p>
</div>

<div style="background-color:#f8f9fb; padding:25px; border-radius:12px; box-shadow:0px 4px 8px rgba(0,0,0,0.08);">

  <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    Introduction:
  </h2>
<p style="text-indent:30px; color: #34495e; font-family: 'Arial', sans-serif; font-size: 15px; margin-top: 18px; line-height:1.6;">
    Epitopes are surface-exposed regions of the antigen that typically form a flat, oblong, oval-shaped area lying above the antibody binding tip (Kringelum et al., 2012). The average plane area is about 400 Ų with a thickness of ~8 Å, and the orientation is typically within a −30° to 60° angle relative to the antibody’s light–heavy chain axis.Each epitope contains roughly 15 amino acid residues, with one linear stretch of 5 or more residues accounting for more than half of its size (Kringelum et al., 2012).
They consist primarily of hydrophobic residues in the center, flanked by charged residues on the periphery (Kringelum et al., 2012). Hydrophilic residues are more evenly distributed but are less common in the hydrophobic core (Kringelum et al., 2012). While no single amino acid is statistically overrepresented compared with equally surface-exposed regions, there is a tendency toward depletion of small hydrophobic residues (Ala, Val, Leu) and enrichment of aromatic residues (Tyr, Trp) (Kringelum et al., 2012). Hydrophobic residues cluster near the antibody contact interface, positively charged residues tend to be further away, and hydrophilic residues occupy intermediate positions (Kringelum et al., 2012). This organization resembles the “wet-rim–dry-core” pattern seen in protein–protein interfaces (Kringelum et al., 2012). Epitopes are loop-rich, with fewer α-helices or β-strands compared to the rest of the antigen surface, reflecting their flexibility and accessibility to antibodies (Kringelum et al., 2012). </p>

<div style="background-color:#f8f9fb; padding:25px; border-radius:12px; box-shadow:0px 4px 8px rgba(0,0,0,0.08);">

  <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    Methods:
  </h2>
  <h3 style="color:#7f8c8d; font-family:'Georgia',serif; font-weight:normal; font-size:22px; margin-top:0;">
    Sequence Used</b>: PDB 7EU8 Chain A & Chain C
  </h3> 
   <p style="color: #34495e; font-family: 'Arial', sans-serif; font-size: 15px; margin-top: 18px;">
    PDB 7EU8 was used as the structural template for solvent-accessibility analysis. This structure represents the human GluN1/GluN2B NMDA receptor in the agonist-bound state, with glycine and glutamate occupying the co-agonist pockets and S-ketamine positioned in the channel vestibule. Within the tetrameric assembly, two GluN1 subunits are present (chains A and C).<br>

Solvent-accessible surface area (SASA) was computed for all residues using DSSP via Biopython, with normalization to per-residue maximum exposure values to derive relative solvent accessibility (RSA). Residues with RSA > 0.3 were classified as surface-exposed, while lower-accessibility residues were considered buried.<br>

This workflow enabled comparative mapping of solvent-exposed, conserved surface loops and strands to predict regions most likely to serve as antibody-accessible epitopes on the NR1 ATD.</p>
</div>

Use VL09 PHOSforUS program

In [14]:
import pandas as pd

# Read the full file into a list of lines
with open("NR1_phosforus.txt") as f:
    lines = f.readlines()

# Find where the Site_wise_details table begins
for i, line in enumerate(lines):
    if line.strip().startswith("Sites"):
        table_start = i
        break

# Extract the table portion
table_lines = lines[table_start:]

# Convert into a pandas DataFrame
# PHOSforUS uses whitespace as delimiter
df = pd.read_csv(
    pd.compat.StringIO("".join(table_lines)),
    sep=r"\s+",
    engine="python"
)
# If your column is still called "Sites", rename it once:
df.rename(columns={"Sites": "Residue"}, inplace=True)

df.head(20)

,Residue,P-type,Predict,Log_score,Phos_prob
0,2,S-,NONPHOS,-2.9308671192467415,0.050649
1,3,T-,NONPHOS,-2.6587913690410745,0.065449
2,8,T-,NONPHOS,-2.2329613177612755,0.096829
3,14,S-,NONPHOS,-2.222427813791195,0.097754
4,16,S-,NONPHOS,-1.9301729388520588,0.126731
5,34,S-,NONPHOS,-1.1988257828446978,0.231684
6,35,T-,NONPHOS,-0.927695623542146,0.283392
7,55,S-,PHOSPHO,1.6193692240897075,0.834708
8,63,T-,NONPHOS,-0.18313319490872482,0.454344
9,64,S-,NONPHOS,-0.6460374313413464,0.343883


In [11]:
# Use DSSP to define loops
from Bio.PDB import MMCIFParser

parser = MMCIFParser()
structure = parser.get_structure("7EU8", "7EU8_DSSP.cif")

model = structure[0]                            # first model
chain = model["A"]                              # use chain A for loop boundaries

residue_ss = {}

for residue in chain:
    res_id = residue.get_id()[1]                 # numeric residue ID
    if "pdbx_PDB_secondary_structure" in residue.xtra:
        ss = residue.xtra["pdbx_PDB_secondary_structure"]
    else:
        ss = "C"   # default to coil if DSSP missing
    residue_ss[res_id] = ss
loop_codes = {"C", "T", "S", " "}

def find_loop_centered_at(position, window=20):
    """
    Find contiguous loop residues in a region ± window around position.
    """
    loop_region = []
    for res in range(position - window, position + window + 1):
        if res in residue_ss and residue_ss[res] in loop_codes:
            loop_region.append(res)
    
    # Now convert to contiguous segments
    segments = []
    current = []
    for r in loop_region:
        if not current:
            current = [r]
        elif r == current[-1] + 1:
            current.append(r)
        else:
            segments.append(current)
            current = [r]
    if current:
        segments.append(current)
    
    # return the segment containing the position
    for seg in segments:
        if position in seg:
            return seg
    return None
    
E101_loop = find_loop_centered_at(101)
N368_loop = find_loop_centered_at(368)

print("E101 loop:", E101_loop)
print("N368 loop:", N368_loop)

#E101_loop = loops[101]
#N368_loop = loops[368]


E101 loop: [81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121]
N368 loop: [348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388]


In [34]:
# Define loop boundaries from DSSP
E101_loop = list(range(81, 122))   # 81–121 inclusive
N368_loop = list(range(348, 389))  # 348–388 inclusive

# Make sure df.Residue is int
df["Residue"] = df["Residue"].astype(int)

# Boolean membership flags
df["in_E101_loop"] = df["Residue"].isin(E101_loop)
df["in_N368_loop"] = df["Residue"].isin(N368_loop)

# Categorical label
def tag_loop(res):
    if res in E101_loop:
        return "E101"
    if res in N368_loop:
        return "N368"
    return "None"

df["Loop"] = df["Residue"].apply(tag_loop)

df.head()

,Residue,P-type,Predict,Log_score,Phos_prob,in_E101_loop,in_N368_loop,Loop
1,2,S-,NONPHOS,-2.9308671192467415,0.05064861469875528,False,False,None
2,3,T-,NONPHOS,-2.6587913690410745,0.0654492212726382,False,False,None
3,8,T-,NONPHOS,-2.2329613177612755,0.09682935421259617,False,False,None
4,14,S-,NONPHOS,-2.222427813791195,0.09775446547702667,False,False,None
5,16,S-,NONPHOS,-1.9301729388520588,0.12673143964323844,False,False,None


In [28]:
df["Residue"] = pd.to_numeric(df["Residue"], errors="coerce")

df = df.dropna(subset=["Residue"])
df["Residue"] = df["Residue"].astype(int)

In [32]:
E101_loop = list(range(81, 122))
N368_loop = list(range(348, 389))

df["in_E101_loop"] = df["Residue"].isin(E101_loop)
df["in_N368_loop"] = df["Residue"].isin(N368_loop)

def tag_loop(res):
    if res in E101_loop:
        return "E101"
    if res in N368_loop:
        return "N368"
    return "None"

df["Loop"] = df["Residue"].apply(tag_loop)

df.head()
df["Loop"].value_counts()
df[df["Loop"] != "None"].head(30)

,Residue,P-type,Predict,Log_score,Phos_prob,in_E101_loop,in_N368_loop,Loop
13,84,S-,NONPHOS,-2.8466955230292434,0.054852381402811214,True,False,E101
14,85,S-,NONPHOS,-2.639171456229305,0.06665956580630579,True,False,E101
15,88,Y-,NONPHOS,-0.6978910424193496,0.3322799766142293,True,False,E101
16,93,S-,NONPHOS,-2.2215258607504045,0.097834045275994,True,False,E101
17,97,TP,NONPHOS,-1.4952790489189272,0.1831306928279698,True,False,E101
18,103,TP,NONPHOS,-1.1652487285649373,0.2377148723969361,True,False,E101
19,105,TP,NONPHOS,-1.045430304348118,0.26010357283763064,True,False,E101
20,108,S-,NONPHOS,-1.5741764721399483,0.17162181743455732,True,False,E101
21,109,Y-,NONPHOS,-0.9969731468123917,0.26953695287522816,True,False,E101
22,110,T-,NONPHOS,-1.918935396305718,0.12798033049689542,True,False,E101


## Quick Interpretation of the above data:

No PHOSPHO predictions appear in the E101 loop. 2 Phospho predictions occur in the N386 loop. 

Many residues show moderate PHOS probability (0.17–0.33 range)

This suggests the LBD hinge loop (N368 loop) is more prone to phosphorylation than the E101 surface loop. 

In [33]:
# Fishers Exact Test to see if this is statistically significant

from scipy.stats import fisher_exact

# Count phosphosite predictions
in_loop = df[df["in_N368_loop"]]
out_loop = df[~df["in_N368_loop"]]

# a = PHOSPHO inside loop
a = (in_loop["Predict"] == "PHOSPHO").sum()

# b = NONPHOS inside loop
b = (in_loop["Predict"] == "NONPHOS").sum()

# c = PHOSPHO outside loop
c = (out_loop["Predict"] == "PHOSPHO").sum()

# d = NONPHOS outside loop
d = (out_loop["Predict"] == "NONPHOS").sum()

# Build contingency table
table = [[a, b],
         [c, d]]

oddsratio, pvalue = fisher_exact(table, alternative="greater")

print("Contingency table:")
print(table)
print("Odds ratio:", oddsratio)
print("P-value:", pvalue)

Contingency table:
[[np.int64(2), np.int64(3)], [np.int64(27), np.int64(106)]]
Odds ratio: 2.617283950617284
P-value: 0.28236638043006906


## Quick interpretation of the Fisher's Exact

Inside N368 loop:
    PHOSPHO = 2
    NONPHOS = 3

Outside N368 loop:
    PHOSPHO = 27
    NONPHOS = 106

Inside loop: 2 / 5 = 40% PHOSPHO

Outside loop: 27 / 133 = 20% PHOSPHO

This is a 2X increase

Odds Ratio = 2.62
This means that a residue inside the N368 loop is ~2.6× more likely to be predicted as phosphosite compared to residues outside

A p-value of ~0.28 is not statistically significant. 
There is a trend toward higher phosphosite propensity in the N368 loop, but it is not statistically significant.

<!-- RESULTS SECTION -->
<div style="background-color:#f8f9fb; padding:25px; border-radius:12px; box-shadow:0px 4px 8px rgba(0,0,0,0.08);">

  <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    Results
  </h2>

  <p style="color:#34495e; font-family:'Arial',sans-serif; font-size:15px; line-height:1.6; text-indent:25px;">
    Using DSSP-based boundaries, the E101 ATD loop spanned residues 81–121, while the N368 LBD hinge loop covered residues 348–388. Mapping PHOSFORUS predictions onto these regions showed that the E101 loop contained multiple serine/threonine residues but none were scored as high-confidence phosphosites. In contrast, the N368 hinge loop contained two PHOSPHO-classified sites (Y351 and Y367) plus several moderately phosphorylatable neighbors.
<br><br>
When enrichment was tested formally, residues inside the N368 loop showed a higher proportion of predicted phosphosites than the rest of NR1 (40% vs ~20%), corresponding to an odds ratio of ~2.6. However, the associated Fisher’s exact p-value (p = 0.28) indicated that this trend did not reach statistical significance. No enrichment was observed in the E101 loop.
  </p>
  <p style="color:#34495e; font-family:'Arial',sans-serif; font-size:15px; line-height:1.6; text-indent:25px;">

<!-- ANALYSIS SECTION -->
<div style="background-color:#f8f9fb; padding:25px; border-radius:12px; box-shadow:0px 4px 8px rgba(0,0,0,0.08);">

  <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    Analysis
  </h2>

  <p style="color:#34495e; font-family:'Arial',sans-serif; font-size:15px; line-height:1.6; text-indent:25px;">
    Overall, phosphorylation predictions do not strongly co-localize with the E101 loop, suggesting that this surface-exposed ATD loop is unlikely to use phosphorylation as a mechanism for epitope masking. The N368 loop, however, behaves differently: it is structurally flexible, functionally important for LBD hinge motion, and the only loop where PHOSFORUS identifies high-confidence phosphosites. The trend toward enrichment (even though not statistically significant) suggests that post-translational modulation of this hinge could influence loop conformation or epitope accessibility.
<br><br>
The computational evidence leans toward possible phosphorylation involvement at the N368 hinge, but the dataset lacks enough positive sites to make a definitive statistical claim. The pattern is suggestive, not conclusive, and requires follow-up with consensus predictors or experimental phospho-proteomics.
  </p>

<!-- REFERENCES SECTION -->
<div style="background-color:#f8f9fb; padding:25px; border-radius:12px; box-shadow:0px 4px 8px rgba(0,0,0,0.08);">

  <h2 style="color:#2c3e50; font-family:'Helvetica Neue',sans-serif; font-size:28px; margin-bottom:20px;">
    References
  </h2>

  <!-- SECTION 1 -->
  <h3 style="color:#7f8c8d; font-family:'Georgia',serif; font-weight:normal; font-size:22px; margin-top:10px;">
    Literature:
  </h3>

  <p style="color:#34495e; font-family:'Arial',sans-serif; font-size:15px; line-height:1.6; text-indent:25px;">
    Tajima, N., Simorowski, N., Yovanno, R. A., Regan, M. C., Michalski, K., Gómez, R., Lau, A. Y., & Furukawa, H. (2022). Development and characterization of functional antibodies targeting NMDA receptors. Nature Communications, 13(1). https://doi.org/10.1038/s41467-022-28559-3<br><br>
      Zhou, C., & Tajima, N. (2023). Structural insights into NMDA receptor pharmacology. Biochemical Society Transactions, 51(4), 1713–1731. https://doi.org/10.1042/bst20230122
      <br><br>
      Wang, H., Xie, C., Deng, B., Ding, J., Li, N., Kou, Z., Jin, M., He, J., Wang, Q., Wen, H., Zhang, J., Zhou, Q., Chen, S., Chen, X., Yuan, T., & Zhu, S. (2024). Structural basis for antibody-mediated NMDA receptor clustering and endocytosis in autoimmune encephalitis. Nature Structural & Molecular Biology. https://doi.org/10.1038/s41594-024-01387-3
  </p>